File: apps/apple/SharedKit/Sources/DesignSystem/Spacing.swift
import SwiftUI

// MARK: - Spacing System
public extension CGFloat {
    struct Spacing {
        public static let xxs: CGFloat = 4
        public static let xs: CGFloat = 8
        public static let sm: CGFloat = 12
        public static let md: CGFloat = 16
        public static let lg: CGFloat = 24
        public static let xl: CGFloat = 32
        public static let xxl: CGFloat = 48
        public static let xxxl: CGFloat = 64
    }
    
    struct Radius {
        public static let xs: CGFloat = 4
        public static let sm: CGFloat = 8
        public static let md: CGFloat = 12
        public static let lg: CGFloat = 16
        public static let xl: CGFloat = 24
        public static let pill: CGFloat = 999
    }
    
    struct IconSize {
        public static let xs: CGFloat = 16
        public static let sm: CGFloat = 20
        public static let md: CGFloat = 24
        public static let lg: CGFloat = 32
        public static let xl: CGFloat = 48
        public static let xxl: CGFloat = 64
    }
}

// MARK: - Spacing Extensions
public extension EdgeInsets {
    static let xxs = EdgeInsets(top: 4, leading: 4, bottom: 4, trailing: 4)
    static let xs = EdgeInsets(top: 8, leading: 8, bottom: 8, trailing: 8)
    static let sm = EdgeInsets(top: 12, leading: 12, bottom: 12, trailing: 12)
    static let md = EdgeInsets(top: 16, leading: 16, bottom: 16, trailing: 16)
    static let lg = EdgeInsets(top: 24, leading: 24, bottom: 24, trailing: 24)
    static let xl = EdgeInsets(top: 32, leading: 32, bottom: 32, trailing: 32)
}

// MARK: - Padding View Extensions
public extension View {
    /// Apply small padding (8pt)
    func paddingXS() -> some View {
        padding(.Spacing.xs)
    }
    
    /// Apply small padding (12pt)
    func paddingSM() -> some View {
        padding(.Spacing.sm)
    }
    
    /// Apply medium padding (16pt)
    func paddingMD() -> some View {
        padding(.Spacing.md)
    }
    
    /// Apply large padding (24pt)
    func paddingLG() -> some View {
        padding(.Spacing.lg)
    }
    
    /// Apply extra large padding (32pt)
    func paddingXL() -> some View {
        padding(.Spacing.xl)
    }
}

File: apps/mobile/src/design-system/spacing.ts
/**
 * Design System - Spacing
 * 
 * Consistent spacing values used throughout the app.
 * Use these instead of hardcoded numbers.
 * 
 * @example
 * import { spacing } from '@/design-system'
 * 
 * <View style={{ padding: spacing.md, gap: spacing.sm }}>
 *   {/* Content */}
 * </View>
 */

// Base spacing unit (4px)
const BASE_UNIT = 4

// Spacing scale
export const spacing = {
  /** 0px - No spacing */
  none: 0,
  
  /** 4px - Minimal spacing between related items */
  xs: BASE_UNIT * 1,
  
  /** 8px - Small spacing, tight layouts */
  sm: BASE_UNIT * 2,
  
  /** 12px - Small-medium spacing */
  md: BASE_UNIT * 3,
  
  /** 16px - Standard spacing (most common) */
  lg: BASE_UNIT * 4,
  
  /** 20px - Medium-large spacing */
  xl: BASE_UNIT * 5,
  
  /** 24px - Large spacing between sections */
  '2xl': BASE_UNIT * 6,
  
  /** 32px - Extra large spacing */
  '3xl': BASE_UNIT * 8,
  
  /** 40px - Maximum spacing */
  '4xl': BASE_UNIT * 10,
  
  /** 48px - Huge spacing for major sections */
  '5xl': BASE_UNIT * 12,
  
  /** 64px - Massive spacing */
  '6xl': BASE_UNIT * 16,
}

// Component-specific spacing
export const componentSpacing = {
  // Buttons
  button: {
    paddingHorizontal: spacing.lg,
    paddingVertical: spacing.md,
    gap: spacing.sm,
  },
  
  buttonSmall: {
    paddingHorizontal: spacing.md,
    paddingVertical: spacing.sm,
    gap: spacing.xs,
  },
  
  buttonLarge: {
    paddingHorizontal: spacing.xl,
    paddingVertical: spacing.lg,
    gap: spacing.sm,
  },
  
  // Cards
  card: {
    padding: spacing.lg,
    gap: spacing.md,
  },
  
  // Lists
  list: {
    gap: spacing.sm,
    itemPadding: spacing.lg,
  },
  
  // Forms
  input: {
    paddingHorizontal: spacing.md,
    paddingVertical: spacing.md,
  },
  
  // Screen
  screen: {
    padding: spacing.lg,
    gap: spacing['2xl'],
  },
  
  // Header
  header: {
    paddingHorizontal: spacing.lg,
    height: 56,
  },
}

// Border radius scale
export const borderRadius = {
  /** 0px - No rounding */
  none: 0,
  
  /** 4px - Slight rounding */
  sm: 4,
  
  /** 8px - Standard rounding (most common) */
  md: 8,
  
  /** 12px - Medium rounding */
  lg: 12,
  
  /** 16px - Large rounding */
  xl: 16,
  
  /** 20px - Extra large rounding */
  '2xl': 20,
  
  /** 24px - Maximum rounding */
  '3xl': 24,
  
  /** 9999px - Full pill shape */
  full: 9999,
}

// Icon sizes
export const iconSize = {
  xs: 12,
  sm: 16,
  md: 20,
  lg: 24,
  xl: 32,
  '2xl': 48,
  '3xl': 64,
}

// Avatar sizes
export const avatarSize = {
  xs: 24,
  sm: 32,
  md: 40,
  lg: 48,
  xl: 64,
  '2xl': 96,
}

// Safe area insets (for screen edges)
export const safeArea = {
  top: 44,    // Status bar height
  bottom: 34, // Home indicator height
  sides: spacing.lg,
}

// Layout utilities
export const layout = {
  maxWidth: {
    sm: 480,
    md: 768,
    lg: 1024,
    xl: 1280,
  },
  minTouchTarget: 44, // Minimum size for interactive elements
}

// Helper functions
export const calculateSpacing = (multiplier: number): number => {
  return BASE_UNIT * multiplier
}

export const rem = (pixels: number): number => {
  return pixels / 16
}
